### ISSUE TITLE
Add git tags for beta versions?

### ISSUE THREAD

**Issue by sunshowers** [rust-lang/rust#100465]
Hi there --

I recently found myself wanting to go back and look at the source of a Rust beta version, and found that unfortunately no tags or releases had been published for them. Wondering if it would be possible to publish tags for beta versions in the future.

I don't particularly care about publishing a GitHub Release. Just a tag is fine, some sort of reference I can go to to look at the source code for beta versions.

I was pointed to this tracker by @jyn514 :) TIA!

**Comment by jyn514** [comment]
@sunshowers there are actually multiple "beta versions" per version number. I currently have two versions installed:
```
; rustup toolchain list | grep beta
beta-2022-05-20-x86_64-unknown-linux-gnu
beta-x86_64-unknown-linux-gnu
; rustc +beta --version
rustc 1.63.0-beta.3 (59f577d63 2022-07-02)
; rustc +beta-2022-05-20-x86_64-unknown-linux-gnu --version
rustc 1.62.0-beta.2 (daf68b1f7 2022-05-19)
```
You can see a list of the manifests for all of them on https://static.rust-lang.org/manifests.txt, and from there the manifest contains a `git_commit_hash` field you can read.

That said, it's a fair bit of work to do that, I think tags are a reasonable ask.

cc @rust-lang/release, what are your thoughts?

**Comment by sunshowers** [comment]
Yeah, absolutely -- ideally when a beta is cut, a tag is published with the version number.

**Comment by jyn514** [comment]
(just to triple check - the version number here would be `1.63.0-beta.3`, not `1.63.0-beta`)

**Comment by sunshowers** [comment]
Yep. I wanted to look at 1.63.0-beta.1 specifically.

**Comment by Mark-Simulacrum** [comment]
There's no technical limitation to doing this, particularly now that promote release is able to tag releases automatically (previously it would have been a manual action and that's too much work). In fact, part of the motivation for writing that code was to allow us to potentially tag beta and nightly releases[^1].

I am not entirely sure, though, whether the tag should be `1.63.0-beta.3`, or `beta-YYYY-MM-DD` -- both of those are kind of "what you want" at different times. I think I would lean towards the first of them, personally, since that's the actual version number, and we've talked about having rustup support installs of specific betas in the past (rather than by date, as today).

**Comment by jyn514** [comment]
I do think it's nice to have the tag match what rustup can install - maybe we should defer adding the tag until we're sure rustup is going to support the `.3` version number in the future?

**Comment by cuviper** [comment]
I would also prefer to have the `beta.N` type of tag.
